client/dashboard/components/settings/general/store-address.tsx

```diff
@@ -348,6 +355,26 @@ export function StoreAddress( {
 		countryState,
 		setValue
 	);
+
+	useEffect( () => {
+		if ( locale ) {
+			storeAddressFields.forEach( ( field ) => {
+				const fieldKey = field
+					.replace( /(address)Line([0-9])/, '$1$2' )
+					.toLowerCase();
+				const props = getInputProps( field );
+				if (
+					locale[ fieldKey ] &&
+					locale[ fieldKey ].hidden &&
+					props.value &&
+					props.value.length > 0
```

Comment:
I am super unfamiliar with TypeScript, but couldn't you do...

```
locale[ fieldKey ]?.hidden &&
props.value?.length > 0
```

Reply:
Yes, most likely, let me add that.

Reply:
@mattsherman updated here: https://github.com/woocommerce/woocommerce-admin/pull/8250/commits/655e1113234b7e25b646288111b73a01e480ec22